extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php

```diff
@@ -0,0 +1,254 @@
+		} else {
+			$payload = $this->decode_token( $token );
+			if ( empty( $payload ) ) {
+				$is_valid_token = false;
+			}
+		}
+
+		if ( $is_valid_token ) {
+			$subscriptions = (array) $payload['subscriptions'];
+		} elseif ( is_user_logged_in() ) {
+			// If there is no token, but the user is logged in, get current subscriptions and determine if the user has
+			// a valid subscription to match the plan ID.
+			$subscriptions = apply_filters( 'earn_get_user_subscriptions_for_site_id', array(), wp_get_current_user()->ID, $this->get_site_id() );
+			if ( empty( $subscriptions ) ) {
+				return false;
+			}
+			// format the subscriptions so that they can be validated.
+			$subscriptions = self::abbreviate_subscriptions( $subscriptions );
+		} else {
+			return false;
+		}
+
+		return $this->validate_subscriptions( $valid_plan_ids, $subscriptions );
+	}
+
+	/**
+	 * Decode the given token.
+	 *
+	 * @param string $token Token to decode.
+	 *
+	 * @return array|false
+	 */
+	public function decode_token( $token ) {
+		try {
+			$key = $this->get_key();
+			return $key ? (array) JWT::decode( $token, $key, array( 'HS256' ) ) : false;
+		} catch ( \Exception $exception ) {
+			// TODO: The SignatureInvalidException is in a different namespace from JWT, so if we want to catch that
+			// one specifically then we'll have to alias it.
+			$logstash = array(
+				'feature' => self::DECODE_EXCEPTION_FEATURE,
+				'message' => self::DECODE_EXCEPTION_MESSAGE,
+				'extra'   => wp_json_encode( compact( 'exception', 'token' ) ),
+			);
+         // phpcs:ignore ImportDetection.Imports.RequireImports.Symbol
+			log2logstash( $logstash );
+			return false;
+		}
+	}
+
+	/**
+	 * Get the key for decoding the auth token.
+	 *
+	 * @return string|false
+	 */
+	abstract public function get_key();
+
+	/**
+	 * Get the ID of the current site.
+	 *
+	 * @return int
+	 */
+	abstract public function get_site_id();
+
+	// phpcs:disable
+	/**
+	 * Get the URL to access the protected content.
+	 *
+	 * @param string $mode Access mode (either "subscribe" or "login").
+	 */
+	public function access_url( $mode = 'subscribe' ) {
+		global $wp;
+		$permalink = get_permalink();
+		if ( empty( $permalink ) ) {
+			$permalink = add_query_arg( $wp->query_vars, home_url( $wp->request ) );
+		}
+
+		$login_url = $this->get_rest_api_token_url( $this->get_site_id(), $permalink );
+		return $login_url;
+	}
+	// phpcs:enable
+
+	/**
+	 * Get the token stored in the auth cookie.
+	 *
+	 * @return ?string
+	 */
+	private function token_from_cookie() {
+		if ( isset( $_COOKIE[ self::JWT_AUTH_TOKEN_COOKIE_NAME ] ) ) {
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			return $_COOKIE[ self::JWT_AUTH_TOKEN_COOKIE_NAME ];
+		}
+	}
+
+	/**
+	 * Store the auth cookie.
+	 *
+	 * @param  string $token Auth token.
+	 * @return void
+	 */
+	private function set_token_cookie( $token ) {
```

Comment:
Since the block is dealing with cookies, that's something we'll need to mention in the Privacy section of the PR so the team working on documentation can include this in the Privacy section of the new support doc.

Reply:
Added